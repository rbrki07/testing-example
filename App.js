// @ts-check
import { StatusBar } from "expo-status-bar";
import React, { useEffect, useState } from "react";
import { SafeAreaView, StyleSheet, Text } from "react-native";
import { InputVariable } from "./components/InputVariable";
import { Operators } from "./components/Operators";
import { calculate } from "./helper/operators";

const App = () => {
  const [variables, setVariables] = useState({ a: NaN, b: NaN });
  const [operator, setOperator] = useState("add");
  const [result, setResult] = useState(NaN);

  useEffect(() => {
    setResult(
      // @ts-ignore
      calculate({ a: variables.a, b: variables.b, operator: operator })
    );
  }, [variables, operator]);

  return (
    <SafeAreaView style={styles.container}>
      <InputVariable
        onSubmitEditing={({ variable }) =>
          setVariables({ ...variables, a: variable })
        }
      />
      <Operators
        // @ts-ignore
        operator={operator}
        setOperator={({ operator }) => setOperator(operator)}
      />
      <InputVariable
        onSubmitEditing={({ variable }) =>
          setVariables({ ...variables, b: variable })
        }
      />
      <Text>{result}</Text>
      <StatusBar style="auto" />
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "flex-start",
  },
});

export default App;
