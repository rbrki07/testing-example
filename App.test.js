// @ts-check
import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import App from "./App";

describe("App integration tests", () => {
  it("should show the correct result", () => {
    const { getAllByPlaceholderText, getByText, queryByText } = render(<App />);
    const inputs = getAllByPlaceholderText("Numeric variable");
    const operator = getByText("+");
    fireEvent(inputs[0], "submitEditing", { nativeEvent: { text: "40" } });
    fireEvent.press(operator);
    fireEvent(inputs[1], "submitEditing", { nativeEvent: { text: "2" } });
    const result = queryByText("42");
    expect(result).toBeOnTheScreen();
  });

  it("should show `NaN` for wrong input", () => {
    const { getAllByPlaceholderText, getByText, queryByText } = render(<App />);
    const inputs = getAllByPlaceholderText("Numeric variable");
    const operator = getByText("+");
    fireEvent(inputs[0], "submitEditing", { nativeEvent: { text: "forty" } });
    fireEvent.press(operator);
    fireEvent(inputs[1], "submitEditing", { nativeEvent: { text: "two" } });
    const result = queryByText("NaN");
    expect(result).toBeOnTheScreen();
  });

  it("should render correctly", () => {
    expect(render(<App />).toJSON()).toMatchSnapshot();
  });
});
