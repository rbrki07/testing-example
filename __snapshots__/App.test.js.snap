// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App integration tests should render correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <View>
    <TextInput
      clearButtonMode="while-editing"
      keyboardType="numeric"
      onSubmitEditing={[Function]}
      placeholder="Numeric variable"
      returnKeyType="done"
      style={
        {
          "backgroundColor": "#f1faee",
          "borderRadius": 8,
          "fontSize": 22,
          "height": 48,
          "margin": 12,
          "paddingLeft": 8,
          "paddingRight": 8,
          "shadowOffset": {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 0.75,
          "shadowRadius": 1,
          "textAlign": "center",
          "width": 192,
        }
      }
    />
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "height": 48,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#f1faee",
          "borderRadius": 8,
          "fontSize": 22,
          "height": 32,
          "justifyContent": "center",
          "margin": 4,
          "opacity": 1,
          "paddingLeft": 4,
          "paddingRight": 4,
          "shadowOffset": {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 0.75,
          "shadowRadius": 1,
          "width": 32,
        }
      }
      testID="addOperator"
    >
      <Text>
        +
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 8,
          "fontSize": 22,
          "height": 32,
          "justifyContent": "center",
          "margin": 4,
          "opacity": 1,
          "paddingLeft": 4,
          "paddingRight": 4,
          "shadowOffset": {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 0.75,
          "shadowRadius": 1,
          "width": 32,
        }
      }
      testID="substractOperator"
    >
      <Text>
        -
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 8,
          "fontSize": 22,
          "height": 32,
          "justifyContent": "center",
          "margin": 4,
          "opacity": 1,
          "paddingLeft": 4,
          "paddingRight": 4,
          "shadowOffset": {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 0.75,
          "shadowRadius": 1,
          "width": 32,
        }
      }
      testID="multiplyOperator"
    >
      <Text>
        *
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 8,
          "fontSize": 22,
          "height": 32,
          "justifyContent": "center",
          "margin": 4,
          "opacity": 1,
          "paddingLeft": 4,
          "paddingRight": 4,
          "shadowOffset": {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 0.75,
          "shadowRadius": 1,
          "width": 32,
        }
      }
      testID="divideOperator"
    >
      <Text>
        /
      </Text>
    </View>
  </View>
  <View>
    <TextInput
      clearButtonMode="while-editing"
      keyboardType="numeric"
      onSubmitEditing={[Function]}
      placeholder="Numeric variable"
      returnKeyType="done"
      style={
        {
          "backgroundColor": "#f1faee",
          "borderRadius": 8,
          "fontSize": 22,
          "height": 48,
          "margin": 12,
          "paddingLeft": 8,
          "paddingRight": 8,
          "shadowOffset": {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 0.75,
          "shadowRadius": 1,
          "textAlign": "center",
          "width": 192,
        }
      }
    />
  </View>
  <Text>
    NaN
  </Text>
</RCTSafeAreaView>
`;
